suite-native/device/src/middlewares/deviceMiddleware.ts

```diff
@@ -63,10 +57,14 @@ export const prepareDeviceMiddleware = createMiddlewareWithExtraDeps(
             dispatch(handleDeviceDisconnect(action.payload));
         }
 
+        const isConnectEnabled = selectIsDeviceConnectFeatureFlagEnabled(getState());
+
         switch (action.type) {
             case DEVICE.CONNECT:
             case DEVICE.CONNECT_UNACQUIRED:
-                dispatch(handleDeviceConnect(action.payload));
+                if (isConnectEnabled) {
```

Comment:
I don't like name `isConnectEnabled` because first thing that popped up in my mind was TrezorConnect and if it's enabled.

Reply:
https://github.com/trezor/trezor-suite/pull/9840/commits/5f26d8a42f7657d579b87b15fda7576d67d39060